### PR TITLE
Add functionality for verifing validated transactions - Closes #880

### LINKED ISSUE #880: Add functionality to verify transactions in transaction pool
### Description

Transaction pool needs to verify transactions from valid queues to move to verified queue.

### PR DESCRIPTION
**Pull request by SargeKhan**
What was the problem?
The functionality for verifying validated transactions was missing.

How did I fix it?
Added functionality for verifying transactions.

How to test it?
Run tests.

Review checklist
The PR resolves #880 
All new code is covered with unit tests
All new code was formatted with Prettier
Linting passes
Tests pass
Commit messages follow the commit guidelines
Documentation has been added/updated

### CODE REVIEW

File: packages/lisk-transaction-pool/test/transaction_pool.ts [hunk 1]
```diff
@@ -454,9 +447,102 @@ describe('transaction pool', () => {
...
+	});
+
+	describe('#verifyValidatedTransactions', () => {
+		const verifiableTransactions = transactions.slice(0, 2);
+		const unverifiableTransactions = transactions.slice(2, 5);
+		const transactionsToVerify = [
+			...verifiableTransactions,
+			...unverifiableTransactions,
+		];
+		let verifyValidatedTransactions: () => Promise<ReadonlyArray<Transaction>>;
```
Comment (shuse2) [inline packages/lisk-transaction-pool/test/transaction_pool.ts]: Prefer to put `let` below `const`

**Review by shuse2 — DISMISSED** [review]
Very minor thing, but otherwise LGTM

**Review by shuse2 — APPROVED** [review]

File: packages/lisk-transaction-pool/src/check_transactions.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+export interface TransactionResponse {
+	readonly errors: ReadonlyArray<Error>;
+	readonly id: string;
+	readonly status: Status;
+}
+
+export enum Status {
+	OK = 1,
+	FAIL,
```
Comment (mitsuaki-u) [inline packages/lisk-transaction-pool/src/check_transactions.ts]: `FAIL = 0, OK = 1`

File: packages/lisk-transaction-pool/src/check_transactions.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	transactions: ReadonlyArray<Transaction>,
+	checkerFunction: CheckerFunction,
+): Promise<CheckTransactionsResponse> => {
+	// Process transactions and check their validity
+	const { transactionsResponses } = await checkerFunction(transactions);
+
+	// Get ids of failed transactions from the response
+	const failedTransactionIds = transactionsResponses
+		.filter(transactionResponse => transactionResponse.status === Status.FAIL)
+		.map(transationStatus => transationStatus.id);
```
Comment (mitsuaki-u) [inline packages/lisk-transaction-pool/src/check_transactions.ts]: typo `transactionStatus`

File: packages/lisk-transaction-pool/src/transaction_pool.ts [hunk 4]
```diff
@@ -236,58 +243,14 @@ export class TransactionPool {
...
 	): boolean {
-		// TODO: remove the condition for checking `verifyTransactionAgainstOtherTransactions` exists, because it should always exist
-		return transaction.verifyTransactionAgainstOtherTransactions
-			? transaction.verifyTransactionAgainstOtherTransactions([
-					...this.queues.ready.transactions,
-					...this.queues.pending.transactions,
-					...this.queues.verified.transactions,
-			  ])
-			: true;
+		return transaction.verifyTransactionAgainstOtherTransactions([
```
Comment (mitsuaki-u) [inline packages/lisk-transaction-pool/src/transaction_pool.ts]: in transactions lib it is named `verifyAgainstOtherTransactions`. the extra `Transaction` seems unneeded

**Review by mitsuaki-u — DISMISSED** [review]
Looks good, just some small name related issues

**Review by mitsuaki-u — APPROVED** [review]

**Pull request merged**